src/NamedDims.jl

```diff
@@ -3,10 +3,31 @@ using Base: @propagate_inbounds
 using Base.Broadcast:
     Broadcasted, BroadcastStyle, DefaultArrayStyle, AbstractArrayStyle, Unknown
 using LinearAlgebra
+using Pkg
+using Requires
 using Statistics
 
 export NamedDimsArray, dim, rename, unname
 
+function __init__()
+    @require Tracker = "9f7883ad-71c0-57eb-9f7f-b5c9e6d3789c" include_tracker_compat()
+end
+
+# For Tracker-NamedDims compatibility we require a Tracker version compatible with 0.2.2.
+# Since we use Requires.jl we have no nicer way to set the Tracker compatability.
+function include_tracker_compat()
+    tracker_version = Pkg.installed()["Tracker"]
```

Comment:
This is not going to work well. `Pkg.installed()` only looks in the Project.toml but it is not guaranteed that Tracker is there if it is just brought in as a dependency.

Reply:
Instead, just declare `Tracker` in `[deps]` and set compat on it with `[compat]` like normally. You can then load it conditionally.

Reply:
> declare Tracker in [deps]

this is what we're trying to avoid

Reply:
I do not know how else to find which version of Tracker is _actually_ installed

Reply:
Oh this is super sad. This will actually break for one of my real used cases too then.
Since i only load `Tracker` via `Flux`

Reply:
ohh, now i see. Gah, silly Pkg not having an easier way to get the installed version of a loaded package. I'll make a PR which ditches this version check (and we hope users are on Tracker v0.2.2, at least for now)

Reply:
Installing a package (without big binary dependencies) is super cheap. It is loading that is expensive. So what is the problem with having to download Tracker and its dependencies? Most of these will already be downloaded already so will be nops.

Reply:
Tracker has a bigish binary dependency from SpecialFuns.jl

Reply:
It's like a 200KB tar download so I would not agree with it being too bigish.